src/upgrade.rs

```diff
@@ -150,7 +150,7 @@ impl Upgraded {
     /// `Upgraded` back.
     pub fn downcast<T: Read + Write + Unpin + 'static>(self) -> Result<Parts<T>, Self> {
         let (io, buf) = self.io.into_inner();
-        match io.__hyper_downcast() {
+        match io.__hyper_downcast::<T>() {
```

Comment:
usually, the convention i am familiar with is to rely on "turbofish" syntax like this when the type parameter is ambiguous, and a specific concrete type must be provided. for example, `iter.collect::<Vec<_>>()`.

my disposition is also to leave this call as is.